### PR TITLE
Add licenses checker for go

### PR DESCRIPTION
**Pull request by mengqiy**

### CODE REVIEW

**Comment by Shell32-Natsu** [comment]
It will be good to have a target in the top level Makefile which checks license for both Go and TS functions.

File: functions/go/Makefile [hunk 1]
```diff
@@ -61,6 +63,29 @@ push: $(FUNCTION_PUSH) ## Push images to registry. WARN: This operation should o
...
 
+.PHONY: check-licenses
+check-licenses: go-licenses $(FUNCTION_CHECKLICENSES)
+
+.PHONY: $(FUNCTION_CHECKLICENSES)
+$(FUNCTION_CHECKLICENSES):
+	$(MAKE) CURRENT_FUNCTION=$(subst -CHECKLICENSES,,$@) func-check-licenses
+
+# find or download go-licenses
+go-licenses:
```
Comment (frankfarzan) [inline functions/go/Makefile:74]: This seems like a hack. Can you just curl the pre-built binary? Note that unlike other tests, you don't need to run this across platforms/arch.

**Review by frankfarzan — CHANGES_REQUESTED** [review]

File: functions/go/Makefile (hunk 1 above)
Comment (mengqiy) [reply functions/go/Makefile:74]: > Can you just curl the pre-built binary?

No. Unfortunately, they don't release pre-built binary.

Using `go get` is their suggested way to install it: https://github.com/google/go-licenses#installation

Comment (frankfarzan) [reply functions/go/Makefile:74]: OK, could you extract lines 76 to 83 into a bash script? ./download_go_license.sh

In general mixing bash and Makefile makes things unreadable.

**Review by frankfarzan — CHANGES_REQUESTED** [review]

**Comment by Shell32-Natsu** [comment]
Will this license check make the CI fail?

**Comment by mengqiy** [comment]
Yes, it has been added to `.github/workflows/ci.yaml`.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**